### ISSUE TITLE
Issues with tofi on river and dual monitor setup

### ISSUE THREAD

**Issue by niels0n** [philj56/tofi#173]
Using [river](https://codeberg.org/river/river), if I launch `tofi` while a window is open and focused on my first monitor, `tofi` spawns on the second monitor instead and I have to go on that monitor, click on it (even if focus should follow cursor so it shouldn't be necessary), and then interact with `tofi`.

If I haven't any open window on the first monitor it works just fine.
On my second monitor this problem doesn't happen, having open windows or not it's irrelevant, and it works even if I have an open window on my first monitor, but the second is focused.

So the problem happen only when `tofi` is launched while focusing a window in the first monitor.
That's weird. This is my config, maybe you can spot something wrong there.

`anchor = center
width = 100%
height = 1440
horizontal = true
font-size = 20
prompt-text = "run: "
font = Tamzen Bold
outline-width = 0
border-width = 0
background-color = #000000
selection-background = #f1c40f
selection-color = #000
selection-padding = 10
input-color=#999
min-input-width = 200
result-spacing = 25
padding-top = 40%
padding-bottom = 0
padding-left = 100
padding-right = 100
margin-left = 0

hide-cursor=true
hint-font = false
text-cursor=true
text-cursor-color=#FFF
drun-launch = true
`

**Comment by redbeardymcgee** [comment]
I'm having the problem exactly as originally described. Version `0.9.1`.

**Comment by ideasman42** [comment]
This is a quick workaround for the issue:

```diff
diff --git a/src/main.c b/src/main.c
index e91838f..b2cd429 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1576,7 +1576,7 @@ int main(int argc, char *argv[])
 	tofi.window.zwlr_layer_surface = zwlr_layer_shell_v1_get_layer_surface(
 			tofi.zwlr_layer_shell,
 			tofi.window.surface.wl_surface,
-			wl_output,
+			nullptr,
 			ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY,
 			"launcher");
 	zwlr_layer_surface_v1_set_keyboard_interactivity(
```

**Comment by riadbelaid** [comment]
Pending a fix, the kludge I've been using relies on setting the `output` configuration option to the currently focused monitor before executing Tofi. Here's my implementation for Sway:

First, add this line to the Tofi configuration:

```
output =
```

Then execute this command (`jq` needs to be installed on the system):

```bash
sed --in-place "s/^output =.*/output = $(swaymsg --type get_outputs | jq '.[] | select(.focused).name')/" ~/.config/tofi/config && tofi-run | xargs swaymsg exec --
```

**Comment by riadbelaid** [comment]
> Didn't work for me on hyprland with hyprctl dispatch instead of swaymsg

Are you executing the command before Tofi on each invocation? E.g.:

```
bindsym Mod4+D exec sed --in-place "s/^output =.*/output = $(swaymsg --type get_outputs | jq '.[] | select(.focused).name')/" ~/.config/tofi/config && tofi-run | xargs swaymsg exec --
```

**Comment by juipeltje** [comment]
i'm still having this issue so i decided to finally look into it and came across this issue. i wouldn't mind trying the workaround that @riadbelaid suggested, but the problem at least on river is that i can't seem to find a command that can list the outputs, or at least i can't seem to find it after reading through the riverctl manpage.


Edit: just managed to find a workaround on river that i wanted to share in case someone finds it helpful. i should note though that i don't use a riverctl next or previous output command to switch between them, instead i have two seperate keybinds that each switch to a specific screen. if you just use a next or previous toggle this probably won't work. so in my river config i execute a script with my usual output keybinds. since river does not support running multiple commands for one keybind in it's config, it has to be done with scripts.

```
# switch focus between outputs
riverctl map normal Super+Control Up spawn ~/.config/river/dp2.sh
riverctl map normal Super+Control Down spawn ~/.config/river/dp1.sh
```

then in the scripts i execute the commands to switch active output, plus change the output setting in the tofi config with sed:

```
#!/bin/sh

# River script for changing both active output and setting output to DP-1 in tofi config
riverctl focus-output DP-1
riverctl spawn "sed -i --follow-symlinks 's/^output.*/output = DP-1/' ~/.config/tofi/config"
```
the other script contains pretty much the same commands, except it has DP-2 instead of DP-1 in it.

**Comment by molarmanful** [comment]
@juipeltje River doesn't have a built-in `swaymsg -t get_outputs` type of thing, *but* there is [river-bedload](https://git.sr.ht/~novakane/river-bedload), which allows you to do:

```bash
tofi-run --output=$(river-bedload -print outputs | jq -r '\''.[] | select(.focused) | .name'\'') | xargs riverctl spawn
```

**Comment by juipeltje** [comment]
@molarmanful yeah i found out about that after making the post but decided not to edit it in anymore lol, had to make a template for it on void linux and it did indeed work like a charm. i ended up switching to fuzzel eventually anyway because tofi also has problems with rendering on smithay.
